### PR TITLE
8261579: AArch64: Support for weaker memory ordering in Atomic

### PR DESCRIPTION
**Pull request by shipilev**
Clean backport to improve AArch64 performance in selected places.

Addtional testing:
 - [x] Linux AArch64 fastdebug build
 - [x] Linux AArch64 fastdebug `tier1`
 - [x] Linux AArch64 fastdebug `tier2`
 - [x] Linux AArch64 release, jcstress `-m quick`

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8261579](https://bugs.openjdk.java.net/browse/JDK-8261579): AArch64: Support for weaker memory ordering in Atomic

### Reviewers
 * [Andrew Haley](https://openjdk.java.net/census#aph) (@theRealAph - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk17u pull/239/head:pull/239` \
`$ git checkout pull/239`

Update a local copy of the PR: \
`$ git checkout pull/239` \
`$ git pull https://git.openjdk.java.net/jdk17u pull/239/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 239`

View PR using the GUI difftool: \
`$ git pr show -t 239`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk17u/pull/239.diff">https://git.openjdk.java.net/jdk17u/pull/239.diff</a>

</details>

### CODE REVIEW

**Comment by shipilev** [comment]
Paging @theRealAph -- do you think this is ready (stable) to go to 17u? This would unblock some Shenandoah backports.

**Comment by theRealAph** [comment]
> Paging @theRealAph -- do you think this is ready (stable) to go to 17u? This would unblock some Shenandoah backports.

This patch is near-zero risk unless you actually use the new weaker functions, so definitely suitable for 17u. The actual uses of these functions require much greater scrutiny.

**Review by theRealAph — APPROVED** [review]

**Comment by shipilev** [comment]
Got the push approval.

/integrate

**Pull request closed without merging**
